modal-go/test/sandbox_mount_image_test.go

```diff
@@ -0,0 +1,177 @@
+package test
+
+import (
+	"context"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/modal-labs/libmodal/modal-go"
+	"github.com/onsi/gomega"
+)
+
+func TestSandboxMountDirectoryEmpty(t *testing.T) {
+	t.Parallel()
+	g := gomega.NewWithT(t)
+	ctx := context.Background()
+	tc := newTestClient(t)
+
+	app, err := tc.Apps.FromName(ctx, "libmodal-test", &modal.AppFromNameParams{CreateIfMissing: true})
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	image := tc.Images.FromRegistry("debian:12-slim", nil)
+
+	sb, err := tc.Sandboxes.Create(ctx, app, image, nil)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	defer terminateSandbox(g, sb)
+
+	mkdirProc, err := sb.Exec(ctx, []string{"mkdir", "-p", "/mnt/empty"}, nil)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	_, err = mkdirProc.Wait(ctx)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	err = sb.ExperimentalMountImage(ctx, "/mnt/empty", nil)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	dirCheck, err := sb.Exec(ctx, []string{"test", "-d", "/mnt/empty"}, nil)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	exitCode, err := dirCheck.Wait(ctx)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	g.Expect(exitCode).To(gomega.Equal(0))
+}
+
+func TestSandboxMountDirectoryWithImage(t *testing.T) {
+	t.Parallel()
+	g := gomega.NewWithT(t)
+	ctx := context.Background()
+	tc := newTestClient(t)
+
+	app, err := tc.Apps.FromName(ctx, "libmodal-test", &modal.AppFromNameParams{CreateIfMissing: true})
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	baseImage := tc.Images.FromRegistry("debian:12-slim", nil)
+
+	sb1, err := tc.Sandboxes.Create(ctx, app, baseImage, nil)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	defer terminateSandbox(g, sb1)
+
+	echoProc, err := sb1.Exec(ctx, []string{
+		"sh",
+		"-c",
+		"echo -n 'mounted content' > /tmp/test.txt",
+	}, nil)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+	_, err = echoProc.Wait(ctx)
+	g.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+	mountImage, err := sb1.SnapshotFilesystem(ctx, 55*time.Second)
```

Comment:
Are we going to advertise that you can snapshot the whole filesystem and then mounting it into `/mnt/data` in another sandbox?